airflow/providers/apache/spark/hooks/spark_submit.py

```diff
@@ -282,10 +313,57 @@ def _resolve_connection(self) -> dict[str, Any]:
     def get_conn(self) -> Any:
         pass
 
+    def _create_keytab_path_from_base64_keytab(self, base64_keytab: str, principal: str | None) -> str:
+        _uuid = uuid.uuid4()
+        temp_dir_path = Path(tempfile.gettempdir()).resolve()
+        temp_file_name = f"airflow_keytab-{principal or _uuid}"
+
+        keytab_path = temp_dir_path / temp_file_name
+        staging_path = temp_dir_path / f".{temp_file_name}.{_uuid}"
```

Comment:
Is there a reason you added here `.{_uuid}`?

Reply:
Yeah! This part was added to assign a unique staging_path to prevent concurrency issues that can occur with the same principal.